src/Libraries/RevitNodes/Elements/Toposolid.cs

```diff
@@ -0,0 +1,373 @@
+        /// <param name="toposolid"></param>
+        private void InternalSetToposolid(Autodesk.Revit.DB.Toposolid toposolid)
+        {
+            InternalToposolid = toposolid;
+            InternalElementId = toposolid.Id;
+            InternalUniqueId = toposolid.UniqueId;
+        }
+
+        #endregion
+
+        #region public properties
+        /// <summary>
+        /// Get Slab Shape Points
+        /// </summary>
+        public IEnumerable<Pt> Points
+        {
+            get
+            {
+                if (this.InternalToposolid.GetSlabShapeEditor() == null)
+                {
+                    throw new Exception(Properties.Resources.InvalidShapeEditor);
+                }
+
+                TransactionManager.Instance.EnsureInTransaction(DocumentManager.Instance.CurrentDBDocument);
+                this.InternalToposolid.GetSlabShapeEditor().Enable();
+                TransactionManager.Instance.TransactionTaskDone();
+
+                List<Pt> points = new List<Pt>();
+                foreach (SlabShapeVertex v in this.InternalToposolid.GetSlabShapeEditor().SlabShapeVertices)
+                {
+                    points.Add(v.Position.ToPoint());
+                }
+                return points;
+            }
+        }
+        #endregion
+
+        #region Public static constructors
+        /// <summary>
+        /// Create a Revit Toposolid given its curve outline, points, type and Level
+        /// </summary>
+        /// <param name="outlineCurves"></param>
+        /// <param name="points"></param>
+        /// <param name="toposolidType"></param>
+        /// <param name="level"></param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        public static Toposolid ByOutlinePointsTypeAndLevel(Curve[] outlineCurves, IEnumerable<Pt> points, ToposolidType toposolidType, Level level)
+        {
+            if (outlineCurves == null)
+            {
+                throw new ArgumentNullException("outlineCurves");
+            }
+
+            var toposolid = ByOutlinePointsTypeAndLevel(PolyCurve.ByJoinedCurves(outlineCurves, 0.001, false), points, toposolidType, level);
+
+            return toposolid;
+        }
+
+        /// <summary>
+        /// Create a Revit Toposolid given its curve outline, points, type and Level
+        /// </summary>
+        /// <param name="outline"></param>
+        /// <param name="points"></param>
+        /// <param name="toposolidType"></param>
+        /// <param name="level"></param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <exception cref="ArgumentException"></exception>
+        public static Toposolid ByOutlinePointsTypeAndLevel(PolyCurve outline, IEnumerable<Pt> points, ToposolidType toposolidType, Level level)
+        {
+            if (outline == null)
+            {
+                throw new ArgumentNullException("outline");
```

Comment:
Why not use `nameof(variable)` instead of static string?

Reply:
Oh, not aware of that keyword. I was just refering to other codes in this repo. Will do some minor changes.